orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/SaveV2PipelineTemplateTask.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipelinetemplate.tasks.v2;
+
+import com.netflix.servo.util.Strings;
+import com.netflix.spinnaker.orca.pipelinetemplate.v2schema.model.V2PipelineTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public interface SaveV2PipelineTemplateTask {
+
+  default void validate(V2PipelineTemplate template) {
+    if (template.getId().contains(".")) {
+      throw new IllegalArgumentException("Pipeline Template IDs cannot have dots");
+    }
+
+    List<String> missingFields = new ArrayList<>();
+    if (template.getMetadata().getName() == null || "".equalsIgnoreCase(template.getMetadata().getName())) {
+      missingFields.add("metadata.name");
+    }
+    if (template.getMetadata().getDescription() == null || "".equalsIgnoreCase(template.getMetadata().getDescription())) {
+      missingFields.add("metadata.description");
+    }
+    if (template.getMetadata().getScopes() == null || template.getMetadata().getScopes().isEmpty()) {
+      missingFields.add("metadata.scopes");
+    }
+    if (template.getSchema() == null || template.getSchema().isEmpty()) {
+      missingFields.add("schema");
+    }
+
+    if (!missingFields.isEmpty()) {
+      throw new IllegalArgumentException("Missing required fields: " + Strings.join(",", missingFields.iterator()));
+    }
+
+    if (!template.getSchema().equals(V2PipelineTemplate.V2_SCHEMA_VERSION)) {
+      throw new IllegalArgumentException(String.format("Invalid schema version: %s", template.getSchema()));
+    }
+
+    if (template.getVariables() != null) {
+      List<String> invalidVariableNames = template.getVariables()
+        .stream()
+        .filter(variable -> variable.getName() != null && variable.getName().contains("-"))
+        .map(V2PipelineTemplate.Variable::getName)
+        .collect(Collectors.toList());
+
+      if (!invalidVariableNames.isEmpty()) {
+        throw new IllegalArgumentException("Variable names cannot include dashes (-)."
```

Comment:
I would connect the validation message with the validation error (the filter above). If someone changes that filter expression above to check for, say `~`, and the error says `can't include dashes`, someone is going to be confused.

Reply:
Agreed, will tie this in with a regex defining the acceptable naming.